.github/workflows/publish_collection.yml

```diff
@@ -30,35 +30,70 @@ jobs:
 
       - name: Build and publish the collection
         shell: bash
+        id: build
         run: |
           set -euxo pipefail
+          OLDVER=$(python3 ./get_collection_version.py)
+          echo ::debug::"Old collection version is $OLDVER"
           # Ensure there is no dest_path before running release_collection.py
           dest_path=/var/tmp/collection
           rm -rf "$dest_path"
           echo ::group::Build Collection
-          python3 ./release_collection.py --debug --dest-path "$dest_path"
+          python3 ./release_collection.py --debug --save-current-changelog --dest-path "$dest_path"
           echo ::endgroup::
+          NEWVER=$(python3 ./get_collection_version.py)
           # We are up to date - exit
           if git diff --quiet; then
             echo ::info No roles have new releases - no collection will be published
+            echo "updated=0" >> $GITHUB_OUTPUT
             exit 0
+          elif [ "$OLDVER" == "$NEWVER" ]; then
+            echo ::error::"Collection version $OLDVER has not changed"
+            exit 1
```

Comment:
Does this mean the pipeline will report an error in this case?

Reply:
In fact, yes. This should not happen. release_collection.py should have updated the collection version, if it has not, I consider it a bug.
Do you have a scenario where not updating the version number is valid?

Reply:
> In fact, yes. This should not happen. release_collection.py should have updated the collection version, if it has not, I consider it a bug. Do you have a scenario where not updating the version number is valid?

ok - just wanted to confirm this is a deliberate error